repos/spack_repo/builtin/packages/py_ase/package.py

```diff
@@ -31,15 +35,53 @@ class PyAse(PythonPackage):
     version("3.15.0", sha256="5e22d961b1311ef4ba2d83527f7cc7448abac8cf9bddd1593bee548459263fe8")
     version("3.13.0", sha256="c4046c50debac28415b36616d79aa28e68ae2cd03c013c2aed6a1e3d465c0ee1")
 
-    depends_on("python@2.6:", type=("build", "run"), when="@:3.15.0")
-    depends_on("python@3.5:", type=("build", "run"), when="@3.18.0:")
-    depends_on("python@3.6:", type=("build", "run"), when="@3.20.0:")
-    depends_on("python@3.8:", type=("build", "run"), when="@3.23.0:")
-    depends_on("py-numpy@1.11.3:", type=("build", "run"))
-    depends_on("py-numpy@1.18.5:", type=("build", "run"), when="@3.23.0:")
-    depends_on("py-matplotlib@2.0.0:", type=("build", "run"))
-    depends_on("py-matplotlib@3.3.4:", type=("build", "run"), when="@3.23.0:")
-    depends_on("py-scipy@0.18.1:", type=("build", "run"))
-    depends_on("py-scipy@1.6.0:", type=("build", "run"), when="@3.23.0:")
-    depends_on("py-flask", type=("build", "run"), when="@:3.18.0")
-    depends_on("py-setuptools", type="build")
+    with when("@3.24.0:"):
+        depends_on("python@3.9:", type=("build", "run"))
+        depends_on("py-numpy@1.19.5:", type=("build", "run"))
+        depends_on("py-scipy@1.6.0:", type=("build", "run"))
+        depends_on("py-matplotlib@3.3.4:", type=("build", "run"))
+        depends_on("py-setuptools@61.0.0:", type="build")
+
+    with when("@3.24.0"):
+        depends_on("python@3.9:", type=("build", "run"))
+        depends_on("py-numpy@1.19.5:", type=("build", "run"))
+        depends_on("py-scipy@1.6.0:", type=("build", "run"))
+        depends_on("py-matplotlib@3.3.4:", type=("build", "run"))
+        depends_on("py-setuptools@61.0.0:", type="build")
+
+    with when("@3.23.0"):
+        depends_on("python@3.8:", type=("build", "run"))
+        depends_on("py-numpy@1.18.5:", type=("build", "run"))
+        depends_on("py-scipy@1.6.0:", type=("build", "run"))
+        depends_on("py-matplotlib@3.3.4:", type=("build", "run"))
+        depends_on("py-setuptools@61.0.0:", type="build")
+
+    with when("@3.21.0:3.21.1"):
```

Comment:
FWIW. *Probably* safe to use the simpler range:

```suggestion
    with when("@3.21"):
```